### PR TITLE
Chore: Improve local setup

### PR DESCRIPTION
**Pull request by Wittiest**
## Details
- Modify the README to improve instructions for local setup
- Remove instructions about local setup with Ollama since the repo doesn't support this at the moment
- Add a warning to PostHog rather than an exception and add a dummy client so that someone can run JACoB without PostHog

### CODE REVIEW

File: src/server/analytics/posthog.ts [hunk 1]
```diff
@@ -3,9 +3,13 @@ import { PostHog } from "posthog-node";
...
 if (!posthogApiKey) {
-  throw new Error("Missing PostHog API key");
+  console.warn("PostHog API key not found, analytics will be disabled");
 }
 
-export const posthogClient = new PostHog(posthogApiKey, {
-  host: "https://app.posthog.com",
-});
+class DummyPostHogClient {
+  capture() {}
```
Comment (kleneway) [inline src/server/analytics/posthog.ts:10]: This PR looks great, thanks for adding this in! One minor thing, the linter is complaining about an empty function here so can you change this to: 
class DummyPostHogClient {
  capture() {
    // do nothing
  }
}

**Review by kleneway — CHANGES_REQUESTED** [review]
Looks great, thanks so much. Can you make the minor change to pass the CI linter step? Thanks!

File: src/server/analytics/posthog.ts (hunk 1 above)
Comment (Wittiest) [reply src/server/analytics/posthog.ts]: Done! Ran linter locally and it passed.

**Comment by kleneway** [comment]
Thanks @Wittiest!!

**Pull request merged**